File: common/python/watcherpayload.py
from azure.storage.queue import QueueMessage
import base64, json

class WatcherPayload:
    def __init__(self, payload_type: str, quality_selection: str, url: str):
        self.PayloadType = payload_type
        self.QualitySelection = quality_selection
        self.Url = url

    @classmethod
    def from_message(cls, msg: QueueMessage):
        jcontent = json.loads(base64.b64decode(msg.content))
        return WatcherPayload(**jcontent)

File: common/ts/WatcherPayload.ts
// copied from watcher/watcher.ts and here for reference only
class ExampleWatcherPayload {
    PayloadType: string;
    QualitySelection: string;
    Url: string;

    constructor(payloadType: string, qualitySelection: string, url: string){
        this.PayloadType = payloadType;
        this.QualitySelection = qualitySelection;
        this.Url = url;
    }
}